src/feature/control/control_getinfo.c

```diff
@@ -325,6 +325,33 @@ getinfo_helper_current_time(control_connection_t *control_conn,
   return 0;
 }
 
+/**
+ * Helps to access directory information for a given
+ * type of consensus.
+ *
+ * returns: int
+ *   -1 in cases of an error
+ *   0 otherwise
+ */
+STATIC int
+getinfo_helper_current_consensus(const char* consensus_type,
+                                 char** answer,
+                                 const char** errmsg)
+{
```

Comment:
Why did you delete the code that checks for a consensus in memory in this refactor?
```
    if (we_want_to_fetch_flavor(get_options(), FLAV_NS)) {
      const cached_dir_t *consensus = dirserv_get_consensus("ns");
      if (consensus)
        *answer = tor_strdup(consensus->dir);
    }
    if (!*answer) { /* try loading it from disk */
```

Reply:
I am checking for memory leaks in the code, I am sorry these were not meant for review. I shall have that code back once I understand what is causing the seg fault and memory leak on mingw and clang.